pkg/apis/notification/v2alpha1/dingtalkconfig_types.go

```diff
@@ -0,0 +1,66 @@
+/*
+Copyright 2020 The KubeSphere Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v2alpha1
+
+import (
```

Comment:
storageClass spec is very flexible, it all depends on the implmentation

Reply:
We'll not actively add more receivers because we already have webhook receiver except a very important and popular one, Just like the Alertmanager says in https://prometheus.io/docs/alerting/latest/configuration/#receiver `We're not actively adding new receivers, we recommend implementing custom notification integrations via the webhook receiver.` 

Use something like storageClass might be a good idea, we can take a look if it can fit into our existing multi-tenant config/receriver pattern. But configs/recerivers have many different parameters, not like the storage's volume, and breaking changes like this is something that should be done in a new API version like v3 and not suitable to make these changes in KubeSphere v3.1 and current Notification Manager release.

Reply:
There is a `parameters` field in storageClass https://github.com/kubernetes/api/blob/master/storage/v1/types.go#L46. It's good for a stable API version, for example  `apis/notifications.kubesphere.io/v2alpha1/notificationreciever` will be the only API we are providing for users, receivers only differ in the spec field.  And hypothetically we need to add a new receiver, all we have to do is adding some code in the controller, the rest will stay the same.

Reply:
@wanjunlei any update?

Reply:
It's better not make big changes like this for now to Notification Manager, no time left actually. We'll discuss this after v3.1 is released

Reply:
Strongly recommend making the changes now, it requires more effort to do it later, and also it's not easy for end-users to migrate their configurations

Reply:
Not much migration efforts, we can add two new CRDs for configs/receivers while keeping existing CRDs for some time and then in a new version deprecate the old CRDs and finally remove them.
The changes may not be finished in 3.1's release schedule per our estimation.